File: backend/src/main/java/br/com/transoft/backend/dto/itinerary/ItineraryPresenter.java
package br.com.transoft.backend.dto.itinerary;

import br.com.transoft.backend.dto.driver.DriverPresenter;
import br.com.transoft.backend.dto.route.RoutePresenter;
import br.com.transoft.backend.constants.ItineraryStatus;
import br.com.transoft.backend.constants.ItineraryType;
import br.com.transoft.backend.dto.vehicle.presenter.VehiclePresenter;

public record ItineraryPresenter(
        String itineraryId,
        ItineraryType type,
        ItineraryStatus status,
        String date,
        String startTime,
        String endTime,
        RoutePresenter route,
        DriverPresenter driver,
        VehiclePresenter vehicle
) {}

File: transoft-mobile-app/app/model/ItineraryPresenter.ts
import DriverPresenter from "./DriverPresenter";
import RoutePresenter from "./RoutePresenter";
import VehiclePresenter from "./VehiclePresenter";

export default interface ItineraryPresenter {
    itineraryId: string;
    type: 'IDA' | 'VOLTA';
    status: 'AGENDADO' | 'EM_ANDAMENTO' | 'CONCLUIDO' | 'CANCELADO';
    date: string
    startTime: string;
    endTime: string;
    route: RoutePresenter;
    driver: DriverPresenter;
    vehicle: VehiclePresenter;
}